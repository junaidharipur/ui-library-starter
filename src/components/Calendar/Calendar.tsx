import * as React from "react";

import "react-calendar/dist/Calendar.css";
import "../../styles/calendar.css";

import cx from "classnames";
import ReactCalender from "react-calendar";
import { Button } from "../Button/Button";

export function Calendar({ value = new Date(), onChange, onRemoveClick, onDoneClick }: CalendarProps) {
    React.useEffect(() => {
        const elems = document.querySelector(".react-calendar__month-view__weekdays");
        const monthTitles: Element[] = Array.from((elems as any)?.children);

        monthTitles.forEach(cur => {
            const titlesElem: Element[] = Array.from((cur as any).children);

            titlesElem.forEach(item => {
                item.textContent = item.textContent?.slice(0, 2) || item.textContent;
            });
        });
    }, []);

    return (
        <div className={cx("shadow-4 inline-block overflow-hidden bg-white rounded-xl")}>
            <ReactCalender value={value} onChange={onChange} />
            <div className="flex items-center justify-between px-6 pt-2 pb-10">
                <Button onClick={onRemoveClick} className="mr-[5px]" fullWidth color="tertiary">
                    Remove
                </Button>
                <Button onClick={onDoneClick} className="ml-[5px]" fullWidth color="primary">
                    Done
                </Button>
            </div>
        </div>
    );
}

interface CalendarProps {
    value?: Date;
    onChange?: (...args: any) => any;
    onRemoveClick?: () => void;
    onDoneClick?: () => void;
}
