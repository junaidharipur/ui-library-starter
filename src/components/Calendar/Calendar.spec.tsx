import * as React from "react";

import { render } from "@testing-library/react";
import { Calendar } from "./Calendar";

test("Calendar component is rendering properly", async () => {
    const { container } = render(<Calendar />);

    expect(container).toMatchSnapshot();
});
