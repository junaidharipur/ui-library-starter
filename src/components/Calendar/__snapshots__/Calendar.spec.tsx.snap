// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component is rendering properly 1`] = `
<div>
  <div
    class="shadow-4 inline-block overflow-hidden bg-white rounded-xl w-[510px]"
  >
    <div
      class="react-calendar"
    >
      <div
        class="react-calendar__navigation"
      >
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
          type="button"
        >
          «
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
          type="button"
        >
          ‹
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__label"
          style="flex-grow: 1;"
          type="button"
        >
          <span
            class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
          >
            November 2023
          </span>
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
          type="button"
        >
          ›
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
          type="button"
        >
          »
        </button>
      </div>
      <div
        class="react-calendar__viewContainer"
      >
        <div
          class="react-calendar__month-view"
        >
          <div
            style="display: flex; align-items: flex-end;"
          >
            <div
              style="flex-grow: 1; width: 100%;"
            >
              <div
                class="react-calendar__month-view__weekdays"
                style="display: flex; flex-wrap: nowrap;"
              >
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Monday"
                    title="Monday"
                  >
                    Mo
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Tuesday"
                    title="Tuesday"
                  >
                    Tu
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Wednesday"
                    title="Wednesday"
                  >
                    We
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Thursday"
                    title="Thursday"
                  >
                    Th
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Friday"
                    title="Friday"
                  >
                    Fr
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Saturday"
                    title="Saturday"
                  >
                    Sa
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Sunday"
                    title="Sunday"
                  >
                    Su
                  </abbr>
                </div>
              </div>
              <div
                class="react-calendar__month-view__days"
                style="display: flex; flex-wrap: wrap;"
              >
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="October 30, 2023"
                  >
                    30
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="October 31, 2023"
                  >
                    31
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 1, 2023"
                  >
                    1
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 2, 2023"
                  >
                    2
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 3, 2023"
                  >
                    3
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 4, 2023"
                  >
                    4
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 5, 2023"
                  >
                    5
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 6, 2023"
                  >
                    6
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 7, 2023"
                  >
                    7
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 8, 2023"
                  >
                    8
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 9, 2023"
                  >
                    9
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 10, 2023"
                  >
                    10
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 11, 2023"
                  >
                    11
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 12, 2023"
                  >
                    12
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 13, 2023"
                  >
                    13
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 14, 2023"
                  >
                    14
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 15, 2023"
                  >
                    15
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 16, 2023"
                  >
                    16
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 17, 2023"
                  >
                    17
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 18, 2023"
                  >
                    18
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 19, 2023"
                  >
                    19
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 20, 2023"
                  >
                    20
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 21, 2023"
                  >
                    21
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 22, 2023"
                  >
                    22
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 23, 2023"
                  >
                    23
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__tile--now react-calendar__tile--active react-calendar__tile--range react-calendar__tile--rangeStart react-calendar__tile--rangeEnd react-calendar__tile--rangeBothEnds react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 24, 2023"
                  >
                    24
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 25, 2023"
                  >
                    25
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 26, 2023"
                  >
                    26
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 27, 2023"
                  >
                    27
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 28, 2023"
                  >
                    28
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 29, 2023"
                  >
                    29
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 30, 2023"
                  >
                    30
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 1, 2023"
                  >
                    1
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 2, 2023"
                  >
                    2
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 3, 2023"
                  >
                    3
                  </abbr>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center justify-between px-6 pt-2 pb-[32px]"
    >
      <button
        class="rounded-md px-7 py-3 border font-medium flex justify-center items-center w-full text-base px-7 py-3 loading-6 bg-dark text-white hover:bg-primary-text cursor-pointer mr-[5px]"
        role="button"
      >
        Remove
      </button>
      <button
        class="rounded-md px-7 py-3 border font-medium flex justify-center items-center w-full text-base px-7 py-3 loading-6 bg-primary text-white hover:bg-blue-dark cursor-pointer ml-[5px]"
        role="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;
